internal/resourcestatus/resourcekey.go

```diff
@@ -81,10 +81,24 @@ func toType(apiVersion, kind string) string {
 	return group + strings.ToLower(kind)
 }
 
+func resourcesDefaultState(bd *fleet.BundleDeployment) string {
+	switch bdState := summary.GetDeploymentState(bd); bdState {
+	// NotReady and Modified BD states are inferred from resource statuses, so it's incorrect to used that to calculate resource states
+	case fleet.NotReady, fleet.Modified:
+		if bd.Status.IncompleteState {
+			return "Unknown"
+		} else {
+			return string(fleet.Ready)
```

Comment:
A comment here could make sense, explaining that this does not mean that the bundle deployment itself is considered ready, but that this is a default state which may well be overwritten by individual resources' `Modified` or `NonReady` statuses. This could be function-level documentation too.